amads/time/meter/break_it_up.py

```diff
@@ -222,22 +181,22 @@ def level_pass(self):
     def advance_step(self, positions_list: list):
         """
         For a start position, and a metrical level expressed as a list of starts,
-        finds the next higher value from those levels.
+        find the next higher value from those levels.
         Used for determining iterative divisions.
         """
         for p in positions_list:
             if p > self.updated_start:
                 duration_to_next_position = p - self.updated_start
                 if self.remaining_length <= duration_to_next_position:
                     self.start_duration_pairs.append(
-                        (self.updated_start, self.remaining_length)
+                        (self.updated_start, round(self.remaining_length, 4))
```

Comment:
Are we sure rounding is safe? Might checking with a tolerance be better?

Reply:
As above.